hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,71 @@
+MONGO_DIR = "/var/lib/mongodb"
+
+# check diskspace
+def disk_space
+  logger.info 'Checking available diskspace in /tmp for upgrade'
+  total_space = `df -k --total --exclude-type=tmpfs`.split("\n")[-1].split[3].to_i
+  mongo_size = File.directory?(MONGO_DIR) ? `du -s  /var/lib/mongodb/`.split[0].to_i : 0
+  if total_space < mongo_size
+    logger.error "There is not enough free space #{total_space} the size of MongoDB database is #{mongo_size}, please add additional space to /tmp and try again, exiting."
+    kafo.class.exit(1)
+  else
+    logger.debug "There is #{total_space} free space on disk, which is more than the size of the MongoDB database of #{mongo_size}, continuing with upgrade"
+  end
+end
+
+def migration
+  export_dir = '/tmp/engine_upgrade'
+  logger.info 'Stopping services except Mongo'
+
+  # Verify to stop services before starting migration
+  response = ask('Are you sure you want to continue? This will stop all the services and will take a while depending on the size of your Mongo database.')
+  if response.downcase != 'y'
+    $stderr.puts '** cancelled **'
+    kafo.class.exit(1)
+  else
+    # Create directories and dump mongoDB
+    Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+    logger.info "Starting mongodump to #{export_dir}"
+    Kafo::Helpers.execute("mkdir -p #{export_dir}/backup")
+    Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+
+    # Move datafiles out of mongoDB directory so it will start
+    logger.info 'Export done, stopping Mongo to remove old datafiles'
+    Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
+    logger.info "Moving contents from #{MONGO_DIR}"
+    Kafo::Helpers.execute("mv #{MONGO_DIR}/* #{export_dir}/backup")
+
+    # Start mongoDB and import the dump
+    logger.info 'Turning on mongo and starting import'
+    Kafo::Helpers.execute('katello-service start --only rh-mongodb34-mongod')
+    Kafo::Helpers.execute("mongod --storageEngine wiredTiger --dbpath #{MONGO_DIR}")
+    Kafo::Helpers.execute("mongorestore --host localhost --db pulp_database --quiet --dir #{MONGO_DIR}")
+
+    # Remove old data files
+    logger.info "Import done, removing old data in #{export_dir}/backup"
+    Kafo::Helpers.execute("rm -rf #{export_dir}/backup")
+
+    # update Hiera to wiredTiger for installer run
+    logger.info 'Changing custom Hiera to use wiredTiger for installer Puppet run.'
+    Kafo::Helpers.execute("sed -i -e 's/run the installer with --upgrade-mongo-storage to upgrade to wiredTiger./Do not remove'/g /etc/foreman-installer/custom-hiera.yaml")
+    Kafo::Helpers.execute("sed -i -e 's/mmapv1/wiredTiger/g' /etc/foreman-installer/custom-hiera.yaml")
+
+    # Create engine upgrade file so we know Mongo is wiredTiger for pre_validation hook.
+    File.open(MONGO_ENGINE_COMPLETE, 'w') do |file|
+      file.write("Mongo storage engine upgraded on #{Time.now}")
+    end
+  end
+end
+
+def engine_upgrade_complete?
```

Comment:
I'd reverse the logic: check if there's a `mmapv1` database present. That way you also handle the case where there is no database yet.